File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int majEle = nums[0];
        int count = 0;

        for(int i=0; i<n; i++){
            if(nums[i] == majEle){
                count++;

            }else{
                count--;
                if(count == 0){
                    majEle = nums[i];
                    count++;
                }
            }
        }

        return majEle;
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:

        l, window = 0, set()

        for r in range(len(nums)):
            if r - l > k:
                window.remove(nums[l])
                l += 1
            if nums[r] in window:
                return True
            window.add(nums[r])

        return False
